### ISSUE TITLE
Bun install incorrect dependency for `c8`

### ISSUE THREAD

**Issue by coderaiser** [oven-sh/bun#12004]
### What version of Bun is running?

1.1.14

### What platform is your computer?

mac os

### What steps can reproduce the bug?

```sh
coderaiser@localcmd:~/putout$ git clone https://github.com/coderaiser/putout && bun i --no-save
coderaiser@localcmd:~/putout$ node_modules/.bin/c8 npm test
/Users/coderaiser/putout/node_modules/c8/node_modules/cliui/build/index.cjs:292
const stripAnsi = require('strip-ansi');
                  ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/coderaiser/putout/node_modules/strip-ansi/index.js from /Users/coderaiser/putout/node_modules/c8/node_modules/cliui/build/index.cjs not supported.
Instead change the require of index.js in /Users/coderaiser/putout/node_modules/c8/node_modules/cliui/build/index.cjs to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/coderaiser/putout/node_modules/c8/node_modules/cliui/build/index.cjs:292:19) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v22.3.0
```

### What is the expected behavior?

Expected `c8` installed correctly

### What do you see instead?

Incorrect dependency `strip-ansi` installed for `c8`

### Additional information

`cliui` requires `strip-ansi@6.0.1`, but it is not installed at all, only version `strip-ansi@7` used, which is required by 🐊**Putout**, it is ESM and everything is failed

**Comment by dylan-conway** [comment]
Having trouble reproducing this bug. Does it happen consistently?

We recently started hoisting aliased packages (`@isaacs/cliui` uses three). Does this error still occur if you delete `node_modules/c8/node_modules/` and reinstall?

**Comment by coderaiser** [comment]
Yes, after removing `node_modules` it works good, here is steps to reproduce inside `putout` repository:

```
npm i lerna@6 -g
lerna bootstrap
bun i
```

**Comment by dylan-conway** [comment]
> Yes, after removing `node_modules` it works good, here is steps to reproduce inside `putout` repository:
> 
> ```
> npm i lerna@6 -g
> lerna bootstrap
> bun i
> ```

Thanks, I can repro this. This is happening because bun is not resetting node_modules after installing with another package manager

**Comment by mixn** [comment]
I’m experiencing the same issue with a new version of [`carbon-now-cli`](https://github.com/mixn/carbon-now-cli) that I trying to get out the door. I think I’m missing something, sorry. You can try the actual prerelease via `bun install -g carbon-now-cli@next`.

![Image](https://github.com/user-attachments/assets/9f1f596a-6b6f-460d-8c89-f8944b7643af)
![Image](https://github.com/user-attachments/assets/838a3697-13e3-4874-b379-f15db59b42b1)
![Image](https://github.com/user-attachments/assets/9fc0f191-f532-4482-8a4a-3aabe1fd3c43)

Thanks! @dylan-conway
